sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -222,8 +265,8 @@ case class AlterViewAsCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
-    if (session.sessionState.catalog.isTemporaryTable(name)) {
-      session.sessionState.catalog.createTempView(name.table, analyzedPlan, overrideIfExists = true)
+    if (session.sessionState.catalog.alterTempViewDefinition(name, analyzedPlan)) {
+      // a local/global temp view has been altered, we are done.
```

Comment:
The previous one is not atomic, and here we need a `createLocalOrGlobalTempView` if we follow the existing style, so I went ahead and make an atomic operation: `alterTempViewDefinition`